### PR TITLE
[MHLO] add aten::gelu op pattern

### PR DESCRIPTION
**Pull request by Yancey1989**
See RFC #999 
This PR added the `aten::gelu` op pattern and move some unit tests from `basic.mlir` to `elementwise.mlir`

Co-authored-by: Jiawei Wu <xremold@gmail.com>
Co-authored-by: Tianyou Guo <tianyou.gty@alibaba-inc.com>
Co-authored-by: Xu Yan <yancey.yx@alibaba-inc.com>
Co-authored-by: Ziheng Jiang <ziheng.jiang@bytedance.com>

### CODE REVIEW

**Comment by byronyi** [comment]
Nit: you don't have to list me as co-author; I have not directly involved in development of this patch :)

File: lib/Conversion/TorchToMhlo/BasicOp.cpp [hunk 1]
```diff
@@ -192,10 +189,8 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
     return success();
   }
 };
-} // namespace
 
 // Binary op legalizations for Mul variants.
-namespace {
```
Comment (ZihengJiang) [inline lib/Conversion/TorchToMhlo/BasicOp.cpp:198]: The redundant anonymous namespace declaration is actually requested by LLVM coding style if I am right: https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

Comment (Yancey1989) [reply lib/Conversion/TorchToMhlo/BasicOp.cpp:198]: I got it, thanks!

**Review by fortianyou — APPROVED** [review]
LGTM

**Review by ZihengJiang — APPROVED** [review]

**Pull request merged**